File: frontend/src/types/file.ts
export interface FileRecord {
  id: number
  organisation?: { orgId: number; name?: string }
  integration?: { id: number; name?: string } | null
  sourceType: string
  originalName: string
  storageKey?: string | null
  contentType?: string | null
  size?: number | null
  status?: string | null
  receivedAt?: string | null
  createdAt?: string
  updatedAt?: string
}

export interface FileCreateRequest {
  sourceType: string
  originalName: string
  storageKey?: string | null
  contentType?: string | null
  size?: number | null
  status?: string | null
  integration?: { id: number } | null
}

export interface FileUpdateRequest {
  originalName?: string
  storageKey?: string | null
  contentType?: string | null
  size?: number | null
  status?: string | null
  integration?: { id: number } | null
}

File: src/main/java/com/taceiq/entity/File.java
package com.taceiq.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import java.time.Instant;

@Entity
@Table(name = "files")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class File {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "org_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler", "users", "roles", "configurations", "integrations", "files"})
    private Organisation organisation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "integration_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Integration integration;

    @Column(name = "source_type", nullable = false)
    private String sourceType;

    @Column(name = "original_name")
    private String originalName;

    @Column(name = "storage_key")
    private String storageKey;

    @Column(name = "content_type")
    private String contentType;

    @Column
    private Long size;

    @Column
    private String status;

    @Column(name = "received_at")
    private Instant receivedAt;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    void onCreate() {
        Instant now = Instant.now();
        this.createdAt = now;
        this.updatedAt = now;
        if (this.receivedAt == null) {
            this.receivedAt = now;
        }
    }

    @PreUpdate
    void onUpdate() {
        this.updatedAt = Instant.now();
    }
}
